docs/DeveloperGuide.md

```diff
@@ -101,11 +101,42 @@ output in your terminal:
 You are now ready to begin developing!
 
 ## Design 
-### Overview
-### Architecture Diagram
+### Architecture Overview
 
 ![Architecture-Diagram](high-level-diagram/architecture_diagram.png)
 
+The architecture diagram above shows the high-level design of the application.
+Given below is a quick overview of the main components of the application
+and their interactions.
+
+#### Main components of the architecture
+- `Main`: The main component that starts the application upon launch of the applicaiton.
+- `WerkIt`: Initializes other components in the correct sequence, and connects them up with each other.
+- `LogHandler`: Handles logging within the application.
+- `Storage`: Reads data from, and writes data to the user's local storage.
+- `UI`: The UI of the application that deals with interaction with the user.
+- `Parser`: Parses user input to make sense of the command supplied by the user.
+- `Command`: Holds the different types of commands available in the application.
+
+#### How the components interact with each other
+[Sequence diagram]
+
+### Component Overview
+
+#### Storage component
+[Writeup]
+#### UI component
+[Writeup]
+#### Parser component
+[Writeup]
+#### Command component
+[Writeup]<br>
+Each command is a feature in the WerkIt! application.
+Thus, the next section will explain the design of each
+features in detail.
+
+### Feature Overview
+
 The features of WerkIt! are split and grouped into 5 main features:
```

Comment:
I just realized that in our DG we classified as 5 main features but in UG we classified as 4. Should we standardize this?